File: leetcode/1491. Average Salary Excluding the Minimum and Maximum Salary/1491.cpp
// https://helloacm.com/function-to-compute-the-average-salary-excluding-the-minimum-and-maximum-salary/
// https://leetcode.com/problems/average-salary-excluding-the-minimum-and-maximum-salary/
// EASY

class Solution {
public:
    double average(vector<int>& salary) {
        auto x = minmax_element(begin(salary), end(salary));
        double sum = std::accumulate(begin(salary), end(salary), 0, [](auto &a, auto &b) {
            return a + b;
        });
        return (sum - *x.first - *x.second) / (salary.size() - 2);
    }
};

File: leetcode/1491. Average Salary Excluding the Minimum and Maximum Salary/1491.py
# https://helloacm.com/teaching-kids-programming-compute-average-of-an-array-excluding-max-and-min/
# https://leetcode.com/problems/average-salary-excluding-the-minimum-and-maximum-salary/
# EASY, MATH

class Solution:
    def average(self, A: List[int]) -> float:       
        maxv, minv = A[0], A[0]
        s = 0
        for i in A:
            maxv = max(maxv, i)
            minv = min(minv, i)
            s += i
        return (s - maxv - minv) / (len(A) - 2)
